mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -277,6 +281,28 @@ private TaskExecutorsList getTaskExecutors(Predicate<Entry<TaskExecutorID, TaskE
                     .collect(Collectors.toList()));
     }
 
+    private void getActiveJobs(GetActiveJobsRequest req) {
+        List<String> pagedList = this.taskExecutorStateMap
+            .values()
+            .stream()
+            .map(TaskExecutorState::getWorkerId)
+            .filter(Objects::nonNull)
+            .map(WorkerId::getJobId)
+            .distinct()
+            .skip(req.getStartingIndex().orElse(0))
```

Comment:
It might be a good idea to do a sort here since the list will not be deterministic otherwise.

Reply:
Good catch!